src/Orleans/SystemTargetInterfaces/IRemoteGrainDirectory.cs

```diff
@@ -22,7 +22,7 @@ internal interface IGrainInfo
         bool AddActivation(ActivationId act, SiloAddress silo);
         ActivationAddress AddSingleActivation(GrainId grain, ActivationId act, SiloAddress silo, GrainDirectoryEntryStatus registrationStatus);
         bool RemoveActivation(ActivationId act, UnregistrationCause cause, out IActivationInfo entry, out bool wasRemoved);
-        bool Merge(GrainId grain, IGrainInfo other);
+        bool Merge(GrainId grain, IGrainInfo other, IInternalGrainFactory grainFactory);
```

Comment:
>IInternalGrainFactory grainFactory [](start = 52, length = 34)

any (simple) way you avoid this dependency on the interface and have the dependency in the constructor of the implementation instead? I wouldn't go overboard if it requires many changes, as it is currently an internal interface only. But still, if it's simple, that would be more logical.